### PR TITLE
examples

### PR DESCRIPTION
**Pull request by riscdanger**
Implementing #6.

### CODE REVIEW

File: examples/bi-directional-rpc.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	ctx := context.TODO()
+	scanner := bufio.NewScanner(os.Stdin)
+
+	fmt.Print(">>> ")
+	for scanner.Scan() {
+		ping := &Ping{Message: scanner.Text()}
+		pong := &Ping{}
+
+		fmt.Println("send: ", ping.Message)
+		res, err := cli.Call(ctx, "selector", ping, pong)
```
Comment (progrium) [inline examples/bi-directional-rpc.go:48]: convention if not using a respondmux is to have an empty selector

File: examples/bi-directional-rpc.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		p := &Ping{}
+		if err := call.Receive(p); err != nil {
+			res.Return(fmt.Errorf("ping err: %+v", err))
+			return
+		}
+
+		res.Return(reverse(p.Message))
+	})
+
+	cli, _ := rpctest.NewPair(h, codec.JSONCodec{})
```
Comment (progrium) [inline examples/bi-directional-rpc.go:29]: this is for tests, maybe examples should avoid using those utilities. i think peer.New should get you what you want

**Comment by progrium** [comment]
great little example, but when i wrote "bi-directional rpc" i meant both client and server can make and handle calls. this isn't just for "callbacks" but also say you connect to a server with the entire intention of letting it make calls to you. harder example to design lol

**Comment by riscdanger** [comment]
I refactored this into a simple cli as I mentioned in #6. Still need to update the rpc example to actually use the rpc stuff.

**Comment by riscdanger** [comment]
The streaming example is currently hanging. I'll look into how to tell the server peer that no more data is coming.

**Pull request closed without merging**